### PR TITLE
FIX: match subdomain with email domain blacklist

### PR DESCRIPTION
**Pull request by techAPJ**
[requested here](https://meta.discourse.org/t/email-domain-blacklist-with-wildcards/28627/9?u=techapj)

### CODE REVIEW

**Comment by ZogStriP** [comment]
build failed?

**Comment by techAPJ** [comment]
Yes, didn't noticed that :disappointed: .. fixed now

File: lib/validators/email_validator.rb [hunk 1]
```diff
@@ -16,9 +16,13 @@ def validate_each(record, attribute, value)
...
-    domains = setting.gsub('.', '\.')
-    regexp = Regexp.new("@(#{domains})", true)
-    value =~ regexp
+    is_blacklisted = false
+    domains = setting.split("|")
+    email_domain = value.split("@").last if value
+    domains.each { |domain|
+      is_blacklisted = true if email_domain && (email_domain == domain || email_domain.end_with?(".#{domain}"))
+    }
+    is_blacklisted
```
Comment (ZogStriP) [inline lib/validators/email_validator.rb]: Hmm, I would have simply updated the regex to something like `Regexp.new("@(.+\.)?(#{domains})", true)`

Comment (techAPJ) [inline lib/validators/email_validator.rb]: Awesome! Updated the code.

**Pull request merged**